code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -276,6 +279,211 @@
 /mob/living/simple_animal/slime/Process_Spacemove(movement_dir = 0, continuous_move = FALSE)
 	return 2
 
+/mob/living/simple_animal/slime/should_atmos_process(datum/gas_mixture/air, exposed_temperature)
+	return TRUE
+
+/mob/living/simple_animal/slime/atmos_expose(datum/gas_mixture/air, exposed_temperature)
+
+	// No need for the rest yet
+	ASSERT_GAS(/datum/gas/hypernoblium, air)
+
+	var/current_gas_pp
+
+	// Hyper-noblium 'reacts' extremely easily with slimes. Any amount at all and they become immune to fire extinguishers and lose all mutation chances.
+	if(air?.gases[/datum/gas/hypernoblium])
+		current_gas_pp = air.get_breath_partial_pressure(air.gases[/datum/gas/hypernoblium][MOLES])
+		if(current_gas_pp)
+			hypernoblium_expose(current_gas_pp, air)
+
+	air.assert_gases(/datum/gas/plasma, /datum/gas/tritium, /datum/gas/bz, /datum/gas/nitrium, /datum/gas/proto_nitrate, /datum/gas/healium, /datum/gas/zauker, /datum/gas/antinoblium)
```

Comment:
I read in code that you should optionally assert them even if just checking if they exist. Since I don't know or like knowing atmos code I was better safe than sorry